lib/mix/lib/mix/tasks/escript.build.ex

```diff
@@ -257,9 +257,25 @@ defmodule Mix.Tasks.Escript.Build do
     end
   end
 
-  defp strip_beam(beam) do
-    {:ok, {_, stripped_beam}} = :beam_lib.strip(beam)
-    stripped_beam
+  defp strip_beam(beam) when is_binary(beam) do
+    {:ok, _, all_chunks} = :beam_lib.all_chunks(beam)
+    ## Filter out abstract code and compile info.
+    filtered_chunks = ['CInf', 'Abst']
+    significant_chunks =
```

Comment:
Please do:

    significant_chunks = for {name, _} = pair <- all_chunks, name not in filtered_chunks, do: pair